paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -263,19 +296,26 @@ def wait_for_deployment(service, deploy_group, git_sha, soa_dir, timeout):
         )
         raise NoInstancesFound
     for cluster in cluster_map.values():
-        cluster['deployed'] = False
+        cluster['deployed'] = 0
     try:
         with Timeout(seconds=timeout):
-            while True:
+            total_instances = sum([len(v["instances"]) for v in cluster_map.values()])
+            with progressbar.ProgressBar(maxval=total_instances) as bar:
                 for cluster, instances in cluster_map.items():
-                    if not cluster_map[cluster]['deployed']:
-                        cluster_map[cluster]['deployed'] = are_instances_deployed(cluster=cluster,
-                                                                                  service=service,
-                                                                                  instances=instances['instances'],
-                                                                                  git_sha=git_sha)
-                if all([cluster['deployed'] for cluster in cluster_map.values()]):
-                    break
-                time.sleep(10)
+                    if cluster_map[cluster]['deployed'] != len(cluster_map[cluster]['instances']):
+                        cluster_map[cluster]['deployed'] = instances_deployed(
+                            cluster=cluster,
+                            service=service,
+                            instances=instances['instances'],
+                            git_sha=git_sha)
+                        if cluster_map[cluster]['deployed'] == len(cluster_map[cluster]['instances']):
+                            instance_csv = ", ".join(cluster_map[cluster]['instances'])
+                            print "Deploy to %s complete! (%s)" % (cluster, instance_csv)
```

Comment:
Yeah, that's a fair point. I like the logging lib because it makes it easy to call things from elsewhere and then filter what comes back. Not a big deal though, happy for this to print.